pkg/cmd/server/start/start_node.go

```diff
@@ -264,12 +264,17 @@ func StartNode(nodeConfig configapi.NodeConfig) error {
 	if err != nil {
 		return err
 	}
+	oclient, _, err := configapi.GetOpenShiftClient(nodeConfig.MasterKubeConfig)
+	if err != nil {
+		return err
+	}
 	glog.Infof("Starting node %s (%s)", config.KubeletServer.HostnameOverride, version.Get().String())
 
-	RunSDNController(config, nodeConfig)
+	registry := osdn.NewOsdnRegistryInterface(oclient, config.Client)
+	endpointFilter := RunSDNController(registry, config, nodeConfig)
```

Comment:
Couldn't the GetOpenShiftClient() and NewOsdnRegistryInterface() still be done in RunSDNController()?  It could then return an error from GetOpenShiftClient() if needed.